common/common.go

```diff
@@ -537,6 +537,25 @@ func (e *multiError) Is(incoming error) bool {
 	return false
 }
 
+// ExcludeError returns a new error without any errors.Is(err)
+// If there are no other errors it will return nil
+func ExcludeError(err, excl error) error {
+	mE, ok := err.(*multiError)
+	if !ok {
+		if errors.Is(err, excl) {
+			return nil
```

Comment:
Thanks.
I did have that thought about how Unwrap is _normally_ used and support for multi-wrapped errors which aren't our own.
I'm thinking that it *should* be viable to handle all errors which support Unwrap the same way.

To answer your question: Nope, I'm explicitly trying to find the remaining errors after acceptable errors are removed. This is a pattern I tend to use a lot, but it's implemented slightly differently in my own version of an error implementation.

I'll swing at it a bit and come back to you.

Reply:
I'm back from the rabbit hole 🐇 🕳️ 
Thanks for the prod to do better. This really needed a properly idiomatic solution consistent with the direction of the go standard library.

```
Common: Improve AppendError and ExcludeError

This change switches from a stateful multiError to caring more about the
Unwrap() []error interface, the same as [go standard
lib](https://github.com/golang/go/blob/go1.21.4/src/errors/wrap.go#L54-L68)

Notably, if we implement Unwrap() []error and do NOT implement Is() then
we get free compatibility with the core functions.

The only distateful thing here is needing to deeply unwrap fmt.Errorf
errors, since they don't flatten. I can't see any way around that
```

Fixed 32f57a0a

Sidenote: I'm left wondering why errors doesn't expose some sort of Unwrappable interface, since both errors and fmt seem to be sharing this inside ~joke~ behaviour 🤷 
Oh well. If it changes tests will tell us.